service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GpcFindDocumentsComponentTest.java

```diff
@@ -0,0 +1,114 @@
+                    .documentId(EhrStatusConstants.DOCUMENT_ID)
+                    .accessDocumentUrl(EhrStatusConstants.GPC_ACCESS_DOCUMENT_URL)
+                    .build()
+            ))
+            .build());
+        return ehrExtractStatusRepository.save(ehrExtractStatus);
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithDocument_Expect_DatabaseToBeUpdated() throws IOException {
+        var ehrExtractStatus = setupDatabase();
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITH_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(taskDefinition.getConversationId()).get();
+        assertThatAccessRecordWasUpdated(updatedEhrExtractStatus, ehrExtractStatus, taskDefinition);
+        Mockito.verify(taskDispatcher).createTask(buildGetDocumentTask(taskDefinition));
+
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithoutDocument_Expect_DatabaseToNotBeUpdated() {
+        var ehrExtractStatus = setupDatabase();
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITHOUT_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(taskDefinition.getConversationId()).get();
+        assertThat(updatedEhrExtractStatus.getGpcAccessDocument().getDocuments().size()).isEqualTo(0);
+
+    }
+
+    private static GpcFindDocumentsTaskDefinition buildFindDocumentTask(EhrExtractStatus ehrExtractStatus, String nhsNumber) {
+        return GpcFindDocumentsTaskDefinition.builder()
+            .fromAsid(ehrExtractStatus.getEhrRequest().getFromAsid())
+            .toAsid(ehrExtractStatus.getEhrRequest().getToAsid())
+            .fromOdsCode(ehrExtractStatus.getEhrRequest().getFromOdsCode())
+            .conversationId(ehrExtractStatus.getConversationId())
+            .requestId(ehrExtractStatus.getEhrRequest().getRequestId())
+            .taskId(UUID.randomUUID().toString())
+            .nhsNumber(nhsNumber)
+            .build();
+    }
+
+    private static GetGpcDocumentTaskDefinition buildGetDocumentTask(GpcFindDocumentsTaskDefinition taskDefinition) {
+        return GetGpcDocumentTaskDefinition.builder()
+            .documentId(GetGpcDocumentTaskDefinition.extractIdFromUrl(VALID_DOCUMENT_URL))
+            .taskId(taskDefinition.getTaskId())
+            .conversationId(taskDefinition.getConversationId())
+            .requestId(taskDefinition.getRequestId())
+            .toAsid(taskDefinition.getToAsid())
+            .fromAsid(taskDefinition.getFromAsid())
+            .fromOdsCode(taskDefinition.getFromOdsCode())
+            .accessDocumentUrl(VALID_DOCUMENT_URL)
+            .build();
+    }
+
+    private void assertThatAccessRecordWasUpdated(EhrExtractStatus ehrExtractStatusUpdated,
```

Comment:
Could move this to the super class since all three component tests use it

Reply:
The structured document test doesn't use this method, mine also asserts different values than the one in getGpcDocument.